src/protagonist/API/Features/NamedQueries/Requests/GetNamedQuery.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using API.Infrastructure.Requests;
+using DLCS.Model.Assets.NamedQueries;
+using DLCS.Repository;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Features.NamedQueries.Requests; 
+
+public class GetNamedQuery : IRequest<FetchEntityResult<NamedQuery>>
+{
+    public int CustomerId { get; }
+    
+    public string NamedQueryId { get; }
+    
+    public GetNamedQuery(int id, string namedQueryId)
+    {
+        CustomerId = id;
+        NamedQueryId = namedQueryId;
+    }
+}
+
+public class GetNamedQueryHandler : IRequestHandler<GetNamedQuery, FetchEntityResult<NamedQuery>>
+{
+    private readonly DlcsContext dbContext;
+    
+    public GetNamedQueryHandler(DlcsContext dbContext)
+    {
+        this.dbContext = dbContext;
+    }
+    
+    public async Task<FetchEntityResult<NamedQuery>> Handle(GetNamedQuery request, CancellationToken cancellationToken)
+    {
+        var namedQuery = await dbContext.NamedQueries
```

Comment:
Can a user fetch a global named query?

Reply:
Good eye! This was a pretty big oversight on my part - fixed in 5303a7d571232528866b3c3b207a8194333d289e